services/management/rds_test.go

```diff
@@ -121,13 +126,72 @@ func TestRDSService(t *testing.T) {
 				AwsSecretKey: secretKey,
 			})
 
-			// TODO: Improve this test. https://jira.percona.com/browse/PMM-4896
-			// In our current testing env with current AWS keys, 2 regions are returning errors but we don't know why for sure
-			// Also, probably we can have more than 1 instance or none. PLEASE UPDATE THIS TESTS !
 			assert.NoError(t, err)
-			t.Logf("%+v", instances)
-			assert.GreaterOrEqualf(t, len(instances.RdsInstances), 1, "Should have at least one instance")
+			assert.Equal(t, len(instances.RdsInstances), 2, "Should have two instances")
+			assert.Equal(t, []*managementpb.DiscoverRDSInstance{
+				{
+					Region:        "us-east-1",
+					Az:            "us-east-1a",
+					InstanceId:    "autotest-aurora-mysql-56",
+					NodeModel:     "db.t2.medium",
+					Address:       "autotest-aurora-mysql-56.cstdx0tr6tzx.us-east-1.rds.amazonaws.com",
+					Port:          3306,
+					Engine:        managementpb.DiscoverRDSEngine_DISCOVER_RDS_MYSQL,
+					EngineVersion: "5.6.10a",
+				},
+				{
+					Region:        "us-west-2",
+					Az:            "us-west-2c",
+					InstanceId:    "autotest-mysql-57",
+					NodeModel:     "db.t2.micro",
+					Address:       "autotest-mysql-57.c3uoaol27cbb.us-west-2.rds.amazonaws.com",
+					Port:          3306,
+					Engine:        managementpb.DiscoverRDSEngine_DISCOVER_RDS_MYSQL,
+					EngineVersion: "5.7.22",
+				},
+			}, instances.RdsInstances)
 		})
+
+		type instance struct {
+			az         string
+			instanceID string
+		}
+
+		for _, tt := range []struct {
+			region    string
+			instances []instance
+		}{
+			{"us-east-1", []instance{{"us-east-1a", "autotest-aurora-mysql-56"}}},
+			{"us-west-2", []instance{{"us-west-2c", "autotest-mysql-57"}}},
+		} {
+			t.Run("discoverRDSRegion", func(t *testing.T) {
+				ctx := logger.Set(context.Background(), t.Name())
+				accessKey, secretKey := tests.GetAWSKeys(t)
+
+				// use given credentials, or default credential chain
+				creds := credentials.NewStaticCredentials(accessKey, secretKey, "")
+				cfg := &aws.Config{
+					CredentialsChainVerboseErrors: aws.Bool(true),
+					Credentials:                   creds,
+					HTTPClient:                    new(http.Client),
+				}
+				sess, err := session.NewSession(cfg)
+				require.NoError(t, err)
+
+				// do not break our API if some AWS region is slow or down
+				ctx, cancel := context.WithTimeout(ctx, awsDiscoverTimeout)
+				defer cancel()
+
+				instances, err := discoverRDSRegion(ctx, sess, tt.region)
+
+				assert.NoError(t, err)
+				require.Equal(t, len(instances), len(tt.instances), "Should have one instance")
+				for i, instance := range tt.instances {
+					assert.Equal(t, instance.az, pointer.GetString(instances[i].AvailabilityZone))
+					assert.Equal(t, instance.instanceID, pointer.GetString(instances[i].DBInstanceIdentifier))
```

Comment:
I assume we compare instances this way to produce better diffs? If yes, I think a comment there will help

Reply:
we compare instances this way because there too much fields that we don't need to compare.